### ISSUE TITLE
Easier to debug locking

### ISSUE THREAD

**Issue by DavidVorick** [NebulousLabs/Sia#222]
It would be nice to have smarter locks for debugging.

For example, Lock() and RLock() that panic if they can't get a lock in 60 seconds. At least then you know a mutex is stuck, and you also know which mutex is stuck. And if something is locked for more than 30 seconds, it calls panic(). Then you know exactly which call to Lock() is causing problems.

This would probably need to be done by implementing a new mutex.

**Comment by DavidVorick** [comment]
I'm really dumb. There was no deadlock at all.

The miner was looking for an update from the state before it began mining. State wasn't posting an update for the genesis block, so the miner thought the target was 0. So instead of deadlocking you just had everyone waiting on the miner to find a 0 block.

\>.<
